### PR TITLE
Introduce rpc url monikers for cli

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

I'm tired of typing `--url blabla...`.

Also, I can't instantly parse which cluster `--url` is pointing to when searching through shell history.

#### Summary of Changes

Introduce sensible defaults named after also sensible `ClusterType`.

Fixes #

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -246,11 +246,11 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
             .short("u")
             .long("url")
-            .value_name("URL")
+            .value_name("URL_OR_MONIKER")
             .takes_value(true)
             .global(true)
-            .validator(is_url)
-            .help("JSON RPC URL for the solana cluster"),
+            .validator(is_url_or_moniker)
+            .help("JSON RPC URL or some monikers (\"mainnet-beta\", \"testnet\", \"devnet\", \"localnet\")"),
```
Comment (ryoqun) [inline cli/src/main.rs:253]: If I mindlessly mapped from `ClusterType`, I would choose `development` for `localnet`. But, `--url development` looks odd to me. So, I picked `localnet` from the program development world. Thus, I also introduced new category term for these: `monikers`.

File: clap-utils/src/input_validators.rs [hunk 2]
```diff
@@ -148,6 +148,33 @@ where
...
+        Err(err) => Err(format!("{}", err)),
+    }
+}
+
+pub fn normalize_to_url_if_moniker(url_or_moniker: &str) -> String {
+    match url_or_moniker {
+        "mainnet-beta" => "http://api.mainnet-beta.solana.com",
+        "testnet" => "http://testnet.solana.com",
+        "devnet" => "http://devnet.solana.com",
+        "localnet" => "http://localhost:8899",
```
Comment (ryoqun) [inline clap-utils/src/input_validators.rs:172]: might worth to add `localnode` (in the context of running various cli commands pointing to mainnet-validator running on the local machine).

Comment (mvines) [reply clap-utils/src/input_validators.rs:172]: How about "localhost" (or "local").

**Review by mvines — COMMENTED** [review]
nice idea!

File: clap-utils/src/input_validators.rs [hunk 3]
```diff
@@ -148,6 +148,33 @@ where
...
+            } else {
+                Err("no host provided".to_string())
+            }
+        }
+        Err(err) => Err(format!("{}", err)),
+    }
+}
+
+pub fn normalize_to_url_if_moniker(url_or_moniker: &str) -> String {
+    match url_or_moniker {
```
Comment (t-nelson) [inline clap-utils/src/input_validators.rs:168]: Nice!  I've been wanting to do this for awhile as well.  WDYT about adding (undocumented) short synonyms, "m", "t", "d", "l"?

**Comment by mvines** [comment]
Oh!  Also `https` for the .solana.com endpoints please.   RPC is insecure enough without having to worry about a network-based MITM

File: clap-utils/src/input_validators.rs (hunk 3 above)
Comment (ryoqun) [reply clap-utils/src/input_validators.rs:168]: nice idea: https://github.com/solana-labs/solana/pull/14409/commits/7f880aafa84d06803ae1756d1fe9cf3cd261d4d4

**Comment by ryoqun** [comment]
> Oh! Also `https` for the .solana.com endpoints please. RPC is insecure enough without having to worry about a network-based MITM

oops! yeah, let's use https.... https://github.com/solana-labs/solana/pull/14409/commits/7f880aafa84d06803ae1756d1fe9cf3cd261d4d4

File: clap-utils/src/input_validators.rs (hunk 2 above)
Comment (ryoqun) [reply clap-utils/src/input_validators.rs]: Yeah, I thought about it. But I wanted some proper term for something local.. But after some thought, I decided to stick to `localhost`. Thanks for obvious suggestion. :)

**Review by mvines — APPROVED** [review]

**Pull request merged**